### PR TITLE
added favorite feature

### PR DESCRIPTION
**Pull request by njackson1-code**
Added favorite capability. Used a through association for users and items. Can access favorites during user's session by current_user.favorites.  

Currently just shows favorites through a star icon. Can change design later, just wanted to get this committed so people who need it can do their stuffs. 

Screen Shot 2020-12-04 at 6 44 50 AM

Screen Shot 2020-12-04 at 6 45 14 AM

### CODE REVIEW

File: app/views/items/favorite.js.erb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+//$(".favBtn")[0].innerHTML = "Product Created";
+alert("here")
```
Comment (amycsong98) [inline app/views/items/favorite.js.erb:2]: ?

File: app/views/items/index.html.erb [hunk 2]
```diff
@@ -45,6 +45,11 @@
           <tr class="item">
             <td class="itemname">
               <%= link_to "#{p.name}", item_path(p) %>
+               <% if user_signed_in? %>
+                <% if current_user.favorites.include? p %>
+                  <%= image_tag("favorite.jpg",size: '10')%>
```
Comment (amycsong98) [inline app/views/items/index.html.erb:50]: can you show how this looks like in your PR description?

File: app/controllers/items_controller.rb [hunk 3]
```diff
@@ -26,6 +26,23 @@ def edit
...
+        format.html { redirect_to @item, notice: 'Successfully Unfavorited.' }
+        format.js
+      else 
+        current_user.favorites << @item
+        format.html { redirect_to @item, notice: 'Successfully Favorited.' }
+        format.js
+
+      end
+    end
+    #redirect_to item_path (@item)
```
Comment (mrubakh) [inline app/controllers/items_controller.rb:43]: try to avoid leaving non-significant comments

File: app/views/items/favorite.js.erb (hunk 1 above)
Comment (mrubakh) [reply app/views/items/favorite.js.erb:2]: hahaha this was probably for debugging --> better to delete this afterwards @njackson1-code

File: app/views/items/index.html.erb (hunk 2 above)
Comment (mrubakh) [reply app/views/items/index.html.erb:50]: +1

File: app/views/items/favorite.js.erb (hunk 1 above)
Comment (mrubakh) [reply app/views/items/favorite.js.erb:2]: also wait... shouldnt this be located in `app/javascript`  and not in `app/views/`?

**Review by amycsong98 — APPROVED** [review]
looks good! thanks noah

**Pull request merged**